### PR TITLE
JBIDE-24491 - allow connection wizard to discover registry url

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java [hunk 1]
```diff
@@ -276,6 +291,17 @@ protected GridLayoutFactory adjustAdvancedCompositeLayout(GridLayoutFactory grid
 		return composite;
 	}
 
+	private void discoverRegistryPressed() {
+		String ret = RegistryProviderModel.getDefault().getRegistryURL(pageModel.getSelectedConnection());
+		if( ret == null ) {
+			System.out.println("Nothing found");
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java]: Could be removed

Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java]: @robstryker nice with extension point et all. Can you please add badly needed unit test?

**Comment by adietish** [comment]
@robstryker nice with extension point et all. Can you please add badly needed unit test?

**Comment by adietish** [comment]
@robstryker I tested this, and I'm not sure that it works really.

steps:
#EXEC: edit existing connection to local cdk
#ASSERT: connection wizard shows up
image
#EXEC: click on "Discover"

Result:
Nothing happens. In the log you see "Not found". 
1. How come that nothing is found?
2. If nothing is found we should definitely tell the user

**Comment by adietish** [comment]
testPR

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java [hunk 2]
```diff
@@ -287,10 +289,14 @@ protected GridLayoutFactory adjustAdvancedCompositeLayout(GridLayoutFactory grid
 		return composite;
 	}
 
-	private void discoverRegistryPressed() {
+	private void discoverRegistryPressed(Shell shell) {
 		String ret = RegistryProviderModel.getDefault().getRegistryURL(pageModel.getSelectedConnection());
 		if( ret != null ) {
 			registryURLObservable.setValue(ret);
+		} else {
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java]: Can we report this to the UI in a more UX way: I'm thinking about reporting an Info status to the databinding framework would be more user friendly

**Comment by adietish** [comment]
testPR

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java [hunk 3]
```diff
@@ -276,6 +289,18 @@ protected GridLayoutFactory adjustAdvancedCompositeLayout(GridLayoutFactory grid
 		return composite;
 	}
 
+	private void discoverRegistryPressed(Shell shell) {
+		String ret = RegistryProviderModel.getDefault().getRegistryURL(pageModel.getSelectedConnection());
+		if( ret != null ) {
+			registryURLObservable.setValue(ret);
+		} else {
+			MessageBox messageBox = new MessageBox(shell, SWT.ICON_ERROR | SWT.OK);
+		    messageBox.setMessage("No registry provider found for the given connection.");
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java]: In my case the user error is that I didnt have the adapter started (launched the cdk from cmd-line). So some hint like "You may have to start your CDK server adapter" would be very helpful to the dummy user (like me)

**Comment by adietish** [comment]
Tried it when I did not have the minishift adapter launched: got informed via message dialog that the lookup failed.
Tried it when I had the minishift adapter launched: the value for the registry url was overwritten. 
To me things look ok, we can still file jiras for possible enhancements:
* inform of failed lookup in the title area (not message dialog). I'm not all sure though if that's preferrable. 
* only insert the registry url if there's no value yet, or ask for confirmation before overwriting.

**Comment by adietish** [comment]
build works for me locally without any errors. Jenkins is red though, failing with p2 dependencies. Merging.

**Pull request merged**